src/TmsIntegrationExample.Example/TmsIntegrationExampleProcessor.cs

```diff
@@ -0,0 +1,63 @@
+﻿// <copyright file="MockProcessor.cs" company="Engaged Technologies">
+//      Copyright (c) Engaged Technologies. All rights reserved.
+//  </copyright>
+// <summary>
+//     implementation class for processing Mock orders/shipments
+// </summary>
+
+namespace TmsIntegrationExample.Example
+{
+    using Microsoft.Extensions.Options;
+    using System.IO;
+    using System.Reflection;
+    using System.Threading.Tasks;
+    using TmsIntegrationExample.Example.Transformers;
+    using TmsIntegrationExample.Services.MockApi;
+    using TmsIntegrationExample.Services.TmsGatewayApi;
+    using TmsIntegrationExample.Services.TmsGatewayApi.Configuration;
+
+    public class TmsIntegrationExampleProcessor : ITmsIntegrationExampleProcessor
+    {
+        private readonly ITmsGatewayClient tmsGatewayClient;
+        private readonly IMockClient mockClient;
+        private readonly TmsGatewayConfiguration tmsGatewayConfiguration;
+
+        public TmsIntegrationExampleProcessor(
+            ITmsGatewayClient tmsGatewayClient, 
+            IMockClient mockClient,
+            IOptions<TmsGatewayConfiguration> tmsGatewayConfiguration)
+        {
+            this.tmsGatewayConfiguration = tmsGatewayConfiguration.Value;
+            this.tmsGatewayClient = tmsGatewayClient;
+            this.mockClient = mockClient;
+        }
+
+        public async Task<int> ProcessAsync()
+        {
+            var ordersResult = await mockClient.GetOrdersAsync(Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), @"MockApi\Orders.json"));
```

Comment:
Would it make sense to handle building the path in an extension method so that it can be called like this?
`@"MockApi\Orders.json".PathFromExecutingAssemblyLocation()`
We're a bit repetitive in our use of Path.Combine, etc, to apply relative pathing to it. I realize that in a real scenario they wouldn't even use this so I can see if we aren't interested in decreasing verbosity here.

That being said, we're blurring the lines some between mocking and stuff that might be useful in the real world. Would it make sense to make this method look as real as possible and inject mock-ish items and behavior?

Reply:
Created new extension class and added method to map the path. please review if it requires any aditional modification?